src/python/pants/backend/python/dependency_inference/module_mapper_test.py

```diff
@@ -779,3 +903,48 @@ def test_issue_15111(rule_runner: RuleRunner) -> None:
             }
         )
     )
+
+
+def test_generate_mappings_from_pattern_matches(rule_runner: RuleRunner) -> None:
```

Comment:
What do you think about writing this as a parametrised test:

```python
@pytest.mark.parametrize(
    ("proj_name", "expected_modules")
    [
        ("google-cloud-hardyhar", ["google.cloud.hardyhar", ...]),
        ...
    ]
)
def test_generate_mappings_from_pattern_matches(proj_name: str, expected_modules: list[str]) -> None:
    assert generate_mappings_from_pattern(proj_name, "bar") == expected_modules
```

I find is a slightly nicer experience because one can see all the failures at once.

(Also, I'd expect the `rule_runner` fixture is only required in tests that actually call methods on it, e.g. it's might not be needed in this one and `test_generate_mappings_from_pattern_no_match`. Have you found otherwise?)

Reply:
ah, parameterized is much better. will do that.

and i think you are correct that the `rule_runner` is not needed. likely a copy-paste thing i did and didn't bother to fix.